realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -387,7 +387,8 @@ private void emitValidateTableMethod(JavaWriter writer) throws IOException {
 
         // verify number of columns
         writer.beginControlFlow("if (table.getColumnCount() != " + metadata.getFields().size() + ")");
-        writer.emitStatement("throw new RealmMigrationNeededException(transaction.getPath(), \"Field count does not match\")");
+        writer.emitStatement("throw new RealmMigrationNeededException(transaction.getPath(), \"Field count does not match - expected %d\")",
```

Comment:
Why not write the current table count as well? Eg. Field count does not match. Was x, expected y? Then you don't have to attach a debugger afterwards to check it :)